sycl/include/CL/sycl/detail/pi.h

```diff
@@ -86,41 +86,10 @@ using pi_native_handle = uintptr_t;
 // TODO: populate PI enums.
 //
 typedef enum {
-  PI_SUCCESS = 0,
-  PI_ERROR_INVALID_KERNEL_NAME = -46,
-  PI_ERROR_INVALID_OPERATION = -59,
-  PI_ERROR_INVALID_KERNEL = -48,
-  PI_ERROR_INVALID_QUEUE_PROPERTIES = -35,
-  PI_ERROR_INVALID_VALUE = -30,
-  PI_ERROR_INVALID_CONTEXT = -34,
-  PI_ERROR_INVALID_PLATFORM = -32,
-  PI_ERROR_INVALID_DEVICE = -33,
-  PI_ERROR_INVALID_BINARY = -42,
-  PI_ERROR_INVALID_QUEUE = -36,
-  PI_ERROR_OUT_OF_HOST_MEMORY = -6,
-  PI_ERROR_INVALID_PROGRAM = -44,
-  PI_ERROR_INVALID_PROGRAM_EXECUTABLE = -45,
-  PI_ERROR_INVALID_SAMPLER = -41,
-  PI_ERROR_INVALID_BUFFER_SIZE = -61,
-  PI_ERROR_INVALID_MEM_OBJECT = -38,
-  PI_ERROR_OUT_OF_RESOURCES = -5,
-  PI_ERROR_INVALID_EVENT = -58,
-  PI_ERROR_INVALID_EVENT_WAIT_LIST = -57,
-  PI_ERROR_MISALIGNED_SUB_BUFFER_OFFSET = -13,
-  PI_ERROR_BUILD_PROGRAM_FAILURE = -11,
-  PI_ERROR_INVALID_WORK_GROUP_SIZE = -54,
-  PI_ERROR_COMPILER_NOT_AVAILABLE = -3,
-  PI_ERROR_PROFILING_INFO_NOT_AVAILABLE = -7,
-  PI_ERROR_DEVICE_NOT_FOUND = -1,
-  PI_ERROR_INVALID_WORK_ITEM_SIZE = -55,
-  PI_ERROR_INVALID_WORK_DIMENSION = -53,
-  PI_ERROR_INVALID_KERNEL_ARGS = -52,
-  PI_ERROR_INVALID_IMAGE_SIZE = -40,
-  PI_ERROR_INVALID_ARG_VALUE = -50,
-  PI_ERROR_INVALID_IMAGE_FORMAT_DESCRIPTOR = -39,
-  PI_ERROR_IMAGE_FORMAT_NOT_SUPPORTED = -10,
-  PI_ERROR_MEM_OBJECT_ALLOCATION_FAILURE = -4,
-  PI_ERROR_LINK_PROGRAM_FAILURE = -17,
+#define _PI_ERRC(NAME, VAL) NAME = VAL,
+#include <CL/sycl/detail/pi_error.def>
+#undef _PI_ERRC
+
   PI_ERROR_PLUGIN_SPECIFIC_ERROR =
       -996, ///< PI_ERROR_PLUGIN_SPECIFIC_ERROR indicates
             ///< that an backend spcific error or
```

Comment:
Why these are left here and not moved to sycl/include/CL/sycl/detail/pi_error.def?

Is it because `stringifyErrorCode` handles these values differently from all other codes?

Reply:
`stringifyErrorCode` was the primary motivation for keeping the custom enums separate. If we want to move it to pi_error.def and still keep the error, we could have another set of cases that have an explicitly set message in addition to the information, e.g. `_PI_ERRC_WITH_MSG(PI_ERROR_PLUGIN_SPECIFIC_ERROR, -996, "The plugin has emitted a backend specific error")`.

Reply:
Sounds like a good option to me.

Reply:
They have been moved to pi_error.def.